src/components/childComponents/ViewResult.jsx

```diff
@@ -524,15 +524,60 @@ export default function ViewResult(props) {
 		let cgeEmployment = await getOutputFileJson(cgeEmploymentId);
 		let cgeEmploymentContent = JSON.parse(cgeEmployment);
 
-		setTables({
-			"totalHouseholdIncome":cgeHouseholdIncomeContent,
-			"domesticSupply":cgeDomesticSupplyContent,
-			"employment":cgeEmploymentContent
-		});
+		return {
+			"totalHouseholdIncome": cgeHouseholdIncomeContent,
+			"domesticSupply": cgeDomesticSupplyContent,
+			"employment": cgeEmploymentContent
+		};
+	};
+
+	const previewCGE = async (executionItemId) => {
+
+		let cgeJson = await getCGEJson(executionItemId);
+		setTables(cgeJson);
 		setResultType("cge");
+		setPreviewOn(true); // open preview
+	};
 
-		// open preview
-		setPreviewOn(true);
+	const downloadCGE = async (executionItemId) => {
+		let cgeJson = await getCGEJson(executionItemId);
+		exportJson(cgeJson, "cge");
```

Comment:
maybe try to put the executionItemId into the file name so new downloads don't have the same filename